File: core/src/main/java/com/core/viewmodel/ViewModelProtocol.java
package com.core.viewmodel;

import io.reactivex.disposables.Disposable;

/**
 * @author volkanhotur
 */

public interface ViewModelProtocol {

    void addDisposable(Disposable disposable);

    void removeDisposable(Disposable disposable);

    void dispose();

    void clear();
}

File: core/src/main/java/com/core/viewmodel/ViewModelProtocol.kt
package com.core.viewmodel

import io.reactivex.disposables.Disposable

/**
 * @author volkanhotur
 */
interface ViewModelProtocol {

    fun addDisposable(disposable: Disposable?)

    fun removeDisposable(disposable: Disposable?)

    fun dispose()

    fun clear()
}